File: v2/snippets/java/common/activationTests.java
package common;

import com.javonet.sdk.*;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import utils.ActivationCredentials;

public class activationTests {

    @Test
    @Tag("integration")
    public void Test_Activation_CorrectCredentials_Returns0() {
        // <Javonet_activate>
        int result = Javonet.activate(ActivationCredentials.yourLicenseKey);
        // </Javonet_activate>
        Assertions.assertEquals(0, result);

        // <Javonet_activate_without_credentials>
        int result2 = Javonet.activate("");
        // </Javonet_activate_without_credentials>
        Assertions.assertEquals(0, result2);
    }
}

File: v2/snippets/cpp/common/activationTests.cpp
#include <gtest/gtest.h>
#include "Javonet.h"
#include "ActivationCredentials.h"

using namespace JavonetNS::Cpp::Sdk;

int main(int argc, char* argv[]) {
	::testing::InitGoogleTest(&argc, argv);
	return RUN_ALL_TESTS();
}

namespace ActivationTests {
	TEST(Integration, Test_Activation_CorrectCredentials_Returns0) {
		remove("javonet.lic");
		// <Javonet_activate>
		auto result = Javonet::Activate(ActivationCredentials::yourLicenseKey);
		// </Javonet_activate>
		EXPECT_EQ((int)0, result);

		// <Javonet_activate_without_credentials>
		auto result2 = Javonet::Activate("");
		// </Javonet_activate_without_credentials>
		EXPECT_EQ((int)0, result2);
	}

}